pkg/store/consul/transaction/transaction.go

```diff
@@ -57,48 +66,67 @@ type Txner interface {
 	Txn(txn api.KVTxnOps, q *api.QueryOptions) (bool, *api.KVTxnResponse, *api.QueryMeta, error)
 }
 
+// MustCommit is a convenience wrapper for Commit that returns a single error
+// if the transaction fails OR if the transaction is rolled back. If the caller
+// wishes to take different action depending on if the transaction failed or
+// was rolled back then Commit() should be used instead
+func MustCommit(ctx context.Context, txner Txner) error {
+	ok, resp, err := Commit(ctx, txner)
+	if err != nil {
+		return err
+	}
+
+	if !ok {
+		return util.Errorf("transaction was rolled back: %s", TxnErrorsToString(resp.Errors))
+	}
+
+	return nil
+}
+
 // Commit attempts to run all of the kv operations in the context's
 // transaction. The cancel function with which the context was created must
 // also be passed to guarantee that the transaction won't be applied twice
-func Commit(ctx context.Context, cancel context.CancelFunc, txner Txner) error {
+func Commit(ctx context.Context, txner Txner) (bool, *api.KVTxnResponse, error) {
```

Comment:
It ends up being more useful in the other PR because there's also a `CommitWithRetries()` function which retries with exponential backoff until a context is canceled, OR there's a transaction conflict. `err != nil` for example could be a TCP error worth retrying, whereas a transaction conflict is unlikely to resolve itself